### ISSUE TITLE
Discover JPA annotations also from overridden super class methods

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7953]
Currently, JPA annotations are used by the `DefaultRecordMapper` and the `DefaultRecordUnmapper` to map methods / attributes to columns when mapping / unmapping. Unfortunately, it is possible to create a situation where the JPA annotations on a super class are not discovered by jOOQ, if a subclass overrides the annotated method:

```java
abstract class Super {
  @Column(name = "Y")
  abstract String getX();
}

class Sub extends Super {
  @Override
  String getX() { ... }
}
```

In the above case, the annotation is not discovered, currently.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.12. Backport scheduled to 3.11.6 (#7984) and 3.10.9 (#7985)

**Comment by lukaseder** [comment]
The fix for this issue has introduced a regression: #8460. Public setters are now called twice. This regression will be fixed in 3.12.0 (#8460), 3.11.11 (#8472) and 3.10.9 (#8473)
